crates/core/tedge_agent/src/state.rs

```diff
@@ -28,7 +29,10 @@ impl StateRepository for AgentStateRepository {
         match fs::read(&self.state_repo_path).await {
             Ok(bytes) => Ok(toml::from_slice::<State>(bytes.as_slice())?),
 
-            Err(err) => Err(StateError::FromIo(err)),
+            Err(err) => {
+                error!("Error reading: {:?}", &self.state_repo_path);
+                return Err(StateError::FromIo(err));
```

Comment:
`return` is unnecessary.

```suggestion
                Err(StateError::FromIo(err))
```